lib/tasks/tranche_maker.rake

```diff
@@ -13,17 +13,18 @@ namespace :tranche do
     beta_test_users = YAML.load_file(Rails.root.join('config/encrypted_private_beta_users.yml'))
 
     new_users.each do |user|
-      encoded_user = user.upcase.tr(' ', '')
+      encoded_user = user.upcase.gsub(' ', '%20')
       uri = URI("https://ccms-pda.legalservices.gov.uk/api/providerDetails/#{encoded_user}")
+      puts uri.to_s
       response = Net::HTTP.get(uri)
       hash = JSON.parse(response)
       if hash.key?('error')
         puts ">>>>>>>> User #{user} / #{encoded_user} not found <<<<<<<<<"
       else
-        contact_hash = hash['contacts'].detect { |h| h['name'] == encoded_user.upcase }
-        whitelisted_users << encoded_user
+        contact_hash = hash['contacts'].detect { |h| h['name'] == user.upcase }
```

Comment:
I can't see how this `==` works?
On line 16 we find spaces and replace them with the encoed space `%20` is the name value in contacts with a normal, unencoded, space? 😕

Reply:
If a user has a space in their username, then the url has to be encoded with %20.  Then in the payload, we look for the username in upper case with a space, not %20 in the list of contacts to get their contact Id.